api-get-object.go

```diff
@@ -105,7 +105,7 @@ func (c Client) GetObject(bucketName, objectName string) (*Object, error) {
 							// Do not set objectInfo from the first readAt request because it will not get
 							// the whole object.
 							reqHeaders.SetRange(req.Offset, req.Offset+int64(len(req.Buffer))-1)
-							httpReader, _, err = c.getObject(bucketName, objectName, reqHeaders)
+							httpReader, objectInfo, err = c.getObject(bucketName, objectName, reqHeaders)
```

Comment:
>Can you test this with all other conditions such as a combination of ReadAt() and then Read()/Seek() ?

I have added a functional test that makes a ReadAt, Stat and ReadAt call on `minio.Object` in that order, in api_functional_v4_test.go. Between the two ReadAt calls the object is modified in the store. I shall see how I can add tests that will cover other sequences of `minio.Object` methods, especially Read, ReadAt and Stat.